Scala/Scala.sublime-syntax

```diff
@@ -1090,11 +1387,15 @@ contexts:
   storage-modifiers:
     - match: '\b(?:private\[\S+\]|protected\[\S+\]|private|protected)\b'
       scope: storage.modifier.access.scala
-    - match: \b(?:abstract|final|lazy|sealed|implicit|override)\b
+    # we treat inline as a hard modifier because it's too convoluted to treat it softly
+    - match: '\b(?:abstract|final|lazy|sealed|implicit|override|using|inline)\b'
+      scope: storage.modifier.other.scala
+    #  soft modifiers
+    - match: '\b(?:infix|opaque|open|transparent)\b(?=\s+(?:private|protected|abstract|final|lazy|sealed|implicit|override|def|val|var|type|given|class|trait|object|enum|case\s+class|case\s+object|inline)\b)'
       scope: storage.modifier.other.scala
 
   # see http://www.scala-lang.org/docu/files/ScalaReference.pdf part 1.3.5-6 (page 18)
-  strings:
+  base-strings:
```

Comment:
I'd suggest to scope strings `meta.string.scala string....` so `meta.string` keeps set within interpolation, which clear string scope. That's common practice accross other default syntaxes. 

At the same time scoping interpolation `${...}` `meta.interpolation` would help aligning with other syntaxes.

Reply:
Iirc, the way we scope strings (particularly with interpolation) was still very much in flux when I wrote this stuff and I haven't gone back and updated it.

Reply:
We can address it in a dedicated follow-up PR as well, as it is not directly related with adding support for Scala 3.